pkg/cloudprovider/providers/rackspace/rackspace_test.go

```diff
@@ -107,6 +107,27 @@ func configFromEnv() (cfg Config, ok bool) {
 	return
 }
 
+func TestParseMetaData(t *testing.T) {
+	_, err := parseMetaData(strings.NewReader(""))
+	if err == nil {
```

Comment:
I'm not sure whether you're allowed to or not, but you can also use gophercloud's [test helpers](https://github.com/rackspace/gophercloud/blob/master/testhelper/convenience.go#L214-L257) for some syntactic sugar. It also allows you to handle mock responses a bit easier